### ISSUE TITLE
Change c.height/ztop/zbottom instance vars in axialExpChngr

### ISSUE THREAD

**Issue by albeanth** [terrapower/armi#844]
c.height, c.ztop, c.zbottom should be local variables and not instance variables. 

https://github.com/terrapower/armi/blob/44f287d5a30b83e64709f7988569b8407c8fdc6d/armi/reactor/converters/axialExpansionChanger.py#L208-L238

**Comment by jakehader** [comment]
Hey @albeanth, can you explain a bit more of the reason for the change here? I'm not quite understanding

**Comment by albeanth** [comment]
@jakehader this originated from a chat I had a few weeks ago with @onufer and @mgjarrett . 

We had realized that the attributes `ztop`, `zbottom`, and `height` on the instances of `Component` were not initialized in the class constructor and only used in `axialExpansionChanger.py`. So I made a ticket to change them from instance variables to local variables. However, in looking at it some more, the attribute `ztop` gets used here https://github.com/terrapower/armi/blob/44f287d5a30b83e64709f7988569b8407c8fdc6d/armi/reactor/converters/axialExpansionChanger.py#L217

where we access the top of the component stored in `linkedComponents[c][0]` - which is critical for component realignment during expansion.

We could swap out `zbottom` and `height` but for consistency I'll just leave it as an attribute that gets initialized here. If we want to 1) initialize the instance variables in the constructor, or 2) reformat the code to not use instance variables we can reopen this ticket.
